processing/src/main/java/org/apache/druid/query/topn/BaseTopNAlgorithm.java

```diff
@@ -41,8 +41,11 @@
 public abstract class BaseTopNAlgorithm<DimValSelector, DimValAggregateStore, Parameters extends TopNParams>
     implements TopNAlgorithm<DimValSelector, Parameters>
 {
-  public static Aggregator[] makeAggregators(Cursor cursor, List<AggregatorFactory> aggregatorSpecs)
+
+  public static Aggregator[] makeAggregators(TopNQuery query, Cursor cursor)
   {
+    query.getAggregatorHelper().addAggregatorMemory();
```

Comment:
nit: I think it would be cleaner if there was a method named like `trackAggregatorMemory` in TopNQuery itself. Something like this:
```
public void trackAggregatorMemory() {
    aggregatorHelper.addAggregatorMemory();
}
```